backend/src/main/java/org/polsl/backend/service/SoftwareService.java

```diff
@@ -42,19 +42,28 @@ public SoftwareService(SoftwareRepository softwareRepository,
     this.computerSetSoftwareRepository = computerSetSoftwareRepository;
   }
 
-  public PaginatedResult<SoftwareListOutputDTO> getAllSoftware() {
-    Iterable<Software> softwares = softwareRepository.findAllByValidToIsNull();
+  public PaginatedResult<SoftwareListOutputDTO> getAllSoftware(Specification<Software> spec) {
     List<SoftwareListOutputDTO> softwareListOutputDTO = new ArrayList<>();
-    for (Software software : softwares) {
-      SoftwareListOutputDTO dto = new SoftwareListOutputDTO();
-      dto.setId(software.getId());
-      dto.setName(software.getName());
-      dto.setInventoryNumber(software.getInventoryNumber());
-      dto.setAvailableKeys(software.getAvailableKeys());
-      dto.setKey(software.getKey());
-      dto.setDuration(software.getDuration());
-      softwareListOutputDTO.add(dto);
+    Iterable<Software> softwares;
```

Comment:
Może dajmy tu software bez 's', konsekwentnie z resztą kodu. W angielskim software to też liczba mnoga 😛

Reply:
ok